terraform/transform_resource.go

```diff
@@ -134,8 +134,34 @@ func (n *graphNodeExpandedResource) DependableName() []string {
 
 // GraphNodeDependent impl.
 func (n *graphNodeExpandedResource) DependentOn() []string {
-	config := &GraphNodeConfigResource{Resource: n.Resource}
-	return config.DependentOn()
+	configNode := &GraphNodeConfigResource{Resource: n.Resource}
+	result := configNode.DependentOn()
+
+	// Walk the variables to find any count-specific variables we depend on.
+	configNode.VarWalk(func(v config.InterpolatedVariable) {
+		rv, ok := v.(*config.ResourceVariable)
+		if !ok {
+			return
+		}
+
+		// We only want ourselves
+		if rv.ResourceId() != n.Resource.Id() {
+			return
+		}
+
+		// If this isn't a multi-access (which shouldn't be allowed but
+		// is verified elsewhere), then we depend on the specific count
+		// of this resource, ignoring ourself (which again should be
+		// validated elsewhere).
+		if rv.Index > -1 {
```

Comment:
Actually nothing really looks wrong there and it looks like we could use `Multi`. I'll change and runs ome tests.